### PR TITLE
Organized the graphql resolvers for the pipeline list view

### PR DESCRIPTION
**Pull request by miguelvelezmj25**

### CODE REVIEW

File: cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/RunsCount/index.tsx [hunk 1]
```diff
@@ -33,11 +33,7 @@ function getRunsCount(pipeline) {
   const program = getProgram(pipeline);
   const totalRuns = objectQuery(program, 'totalRuns');
 
-  if (totalRuns === null || totalRuns === undefined) {
-    return 0;
-  }
-
-  return totalRuns;
+  return totalRuns || 0;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/RunsCount/index.tsx]: Was this change intended to be in this pull request?

File: cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js [hunk 2]
```diff
@@ -20,14 +20,14 @@ import { GLOBALS } from 'services/global-constants';
 function getProgram(pipeline) {
   const programs = objectQuery(pipeline, 'applicationDetail', 'programs');
 
-  if (programs === null || programs === undefined) {
-    return [];
+  if (!programs) {
+    return {};
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js]: Any specific reason `programs` returning `{}` instead of `[]`?

File: cdap-ui/app/cdap/services/StatusFactory.js [hunk 3]
```diff
@@ -108,7 +108,7 @@ const getRequestInfo = () => {
   };
   let requestInfo = {
     credentials: 'include',
-    mode: 'no-cors',
+    // mode: 'no-cors',
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/services/StatusFactory.js]: Same here. Was this change intended to be in this pull request?

**Review by ajainarayanan — COMMENTED** [review]
@miguelvelezmj25 One suggestion would be to have named functions in each of the exported resolver files. Would be helpful to debug if at somepoint any error happens. The function name will be available in the stacktrace which is not available in anonymous function.

File: cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/RunsCount/index.tsx (hunk 1 above)
Comment (miguelvelezmj25) [reply cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/RunsCount/index.tsx]: Yes, it came from rebasing with the branch of https://github.com/cdapio/cdap/pull/11486

File: cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js (hunk 2 above)
Comment (miguelvelezmj25) [reply cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js]: Because this function returns A program, not a list of programs.

File: cdap-ui/app/cdap/services/StatusFactory.js (hunk 3 above)
Comment (miguelvelezmj25) [reply cdap-ui/app/cdap/services/StatusFactory.js]: No. I reverted this change.

File: cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js]: Hhmm should this variable then be `program`?

**Review by ajainarayanan — APPROVED** [review]
👍 LGTM

File: cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js (hunk 2 above)
Comment (miguelvelezmj25) [reply cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/graphqlHelper.js]: No, because we first get all programs and then we get the program with the name based on the type of artifact.
